mcs/class/System.ServiceModel.Web/Test/System/UriTemplateTest.cs

```diff
@@ -462,6 +462,19 @@ public void MatchWildcard3 ()
 		}
 
 		[Test]
+		public void MatchIgnoreQueryParamNoValue ()
+		{
+			try
+			{
+				var t = new UriTemplate ("/{a}/*", true);
+				var m = t.Match (new Uri ("http://s"), new Uri ("http://s/a/b?foo"));
+				Assert.AreEqual (1, m.QueryParameters.Keys.Count, "#1");
+			} catch (IndexOutOfRangeException) {
```

Comment:
No need for a try-catch here, as if an exception is thrown, the test will fail anyway.